### DISCUSSION TITLE
pH Pump / Container Calibration

### DISCUSSION THREAD

**Discussion by orrious** [tagyoureit/nodejs-poolController#557] [Nixie]
Been busy this weekend installing new probes, etc.  Pictures to follow.  That said I moved to REM Chem and really enjoying it!

What are your thoughts around "Calibrate Pump" for the pH or liquid chlorine dispenser that would dispense for exactly a minute.  That way you don't have to worry about timing it with a stop watch.  Also a "Prime" button, so you could prime it as well.  Currently folks need to go into REM to run the pump manually for a minute.

I also added a non-contact liquid level sensor on my tank at 1 gal.  Any interest in adding a hardware input for the tank?

**Comment by rstrouse** [comment]
I went kicking and screaming into the REM Chem.  But try to take it away now and I would put up a fight.

I have a timed manual dose on the list of futures for REM Chem.  For the most part the only time you are likely to use it is when you are calibrating.  I have only ever done that once but assume I will do it again when I finally get around to replacing the tube in the pump.

I think you will find that a moderately calibrated pump will give you a very accurate display of the tank level.  I think @tagyoureit tried a non-contact sensor and abandoned it.  They are probably great for black water tanks on RVs.  But I remember folks discussing this at length at one point in the gitter with not-so-great results.  You will be surprised at how remarkably accurate the draw is from the tank due to the way peristaltic pump works.  The design provides highly accurate liquid dispensing for everything from medicines to cleaner formulations.

Almost always you will be adding a gallon at a time to your tank so simply adding 1x the number of gallons to the tanks level when you add it will keep it in sync.  The tank alarm will let you know when it is getting low so you can get some more before you are out.  I buy 4 gallons at a time and dump 3 in the tank with 1 gallon reserve when the acid tank low alarm goes off.  All of this is remarkably no fiddle after you are set up.

**Reply by johnny2678** [reply]
Hold up. You are only putting acid in the tanks? 31ish % MA? No dilution with water?

**Reply by rstrouse** [reply]
34% MA is full strength anything above that s hydrochloric.  I buy 14.5% in the 2 pack from Home Depot.  If you are using 31% add 1 gallon of MA and one gallon of water.   Then set the acid strength to 14.5% in REM Chem.

**Reply by johnny2678** [reply]
Ahh. K. Though you might be going full 31.45 and was curious how it was working for you.  

I do 1:1 with water so I have 15.7% selected in Chem settings

**Reply by tagyoureit** [reply]
I used the contactless sensor for a while.  1 out of 3 went bad, the 2nd only seems to work part time no matter what, and the third works better I pour some water on the outside of the tank where the sensor is.  I have a suspicion that it has never made a really good connection on the round tanks with thick walls.  It would probably work better with thinner material and a flat surface.

What did you use to make connect the sensor to your tank?

But I 100% agree with @rstrouse.  The pumps are very accurate as are the mechanics for tracking the tank level.  I've never had a concern about the app tracking the tank level.  I still have the sensors on the outside of the tanks but they aren't connected to anything in my stack at the moment.

**Reply by orrious** [reply]
I used this non-contact sensor.  https://www.amazon.com/Taidacent-Non-contact-Electronic-Detector-Detection/dp/B07Z64CSLQ

I installed the sensor before I mounted the tank and was able to lay the tank on it's side.   The acid tank is HDPE, so I sanded and flame treated a 2" x 2" square on the 1 gallon mark.  I mixed and let the JB weld plastic epoxy set up a little, so it wasn't so runny.  Then poured a quarter sized puddle on the tank and the back of the sensor and pressed them together.  I held it for 15 minutes by hand and it setup in a nice pad with no voids.  I.e it totally filled the gap between the tank and sensor.  I then mixed another batch and poured over the entire sensor to ensure it was 100% encapsulated in epoxy.  If I even want to replace this, it will be with an angle grinder and flapper pad. 

It so far appears to be very accurate, but time will tell.  I use similar sensors in my other hobbies.

**Reply by orrious** [reply]
![Between pours](https://user-images.githubusercontent.com/9359564/181039519-664b251c-a2f9-4083-a231-4439a60ba3c9.jpeg)

**Comment by johnny2678** [comment]
> Been busy this weekend installing new probes, etc. Pictures to follow. 

Looking forward to it! Industrial probes? Will be curious to hear about your flow cell tapping if so.

**Reply by orrious** [reply]
I had a nice piece of crystal acrylic ready to go, but I’m not one to reinvent the wheel if I don’t have to, so figured I’d try modifying the flow cell first. 

Chucked up the flow cell and used the 59/64” drill as a guide for centering the flow cell hole.  Once locked in position, replaced the 59/64” with a step bit. Used the step bit to reorient the hole to the new center. Swapped back over to the 59/64” bit to bring it to the correct diameter. It cuts easily and if done right, there is just enough room. 

![6F47303B-6BEF-4CA4-B790-790566D0880E](https://user-images.githubusercontent.com/9359564/181041358-48cf7298-0c5c-463e-9d04-bf75746afe6e.jpeg)
![3C014FBE-5047-4111-AE7B-F19939ABEDF2](https://user-images.githubusercontent.com/9359564/181041371-91f9b2af-d5e1-4fde-add5-a2b8b7b0dcf9.jpeg)
![56D48505-63F9-4854-B810-1C883F74E5FD](https://user-images.githubusercontent.com/9359564/181041378-c4213c6e-f2ff-41ec-ba08-ef241c282b4d.jpeg)
![9BA16DB6-3475-4C2B-B6CF-055B77549CA5](https://user-images.githubusercontent.com/9359564/181041385-3e958bfc-202b-4d6a-9538-f6541d9da3fd.jpeg)

**Reply by rstrouse** [reply]
What a great idea to use a step bit.  Cutting that way will certainly eliminate any chatter on the bit.

**Reply by orrious** [reply]
Btw, if anyone is curious what is going on in the picture, it was 98 degrees out. Makes it really hard to calibrate temperate dependent solutions and probes at 25C/77F.  

I have no way of cooling water to a set temp, but I do have a way to heat it to a set point (sous vide / immersion heater). 

So I iced my yeti style cooler down for an hour and let it get as cold as possible and then when ready, the sous vide brought it up to 77 and kept it there.  I did need to add ice to the water a couple times during the process. 

Then just keep everything in the water while calibrating. Obviously don’t let the cooler water into your calibration solutions.

**Reply by rstrouse** [reply]
> So I iced my yeti style cooler down for an hour and let it get as cold as possible and then when ready, the sous vide brought it up to 77 and kept it there. I did need to add ice to the water a couple times during the process.

Funny I just thought you were placing an organ transplant, like a kidney, or liver into the pool.

**Reply by johnny2678** [reply]
LOL - I used my sous vide too.  Great minds

**Reply by johnny2678** [reply]
I don't suppose you took a video of recentering / expanding/ tapping the flow cell?  I really want to try this so I can move up to industrial but worried about messing up and damaging the cell.

**Comment by orrious** [comment]
BTW, calibrating my acid pump was a complete pain.  Took me 2 or 3 goes to finally figure out that njsPC will override REM.  So if you go into REM and manually control the pump, njsPC will turn it off.  Therefor had to go into njsPC and unconfigure the acid pump, then back to REM, time it, etc.  

From a user experience, pressing a single "Calibrate" button that dispenses for exactly 60, would have been more pleasant for me.

**Reply by rstrouse** [reply]
I hear ya it is on the list.

**Reply by tagyoureit** [reply]
You could just stop njsPC for a couple of mins.  Then you don't have to fight it and/or reconfigure it.

**Reply by orrious** [reply]
After the fact knowledge.  Fumbling around trying to hold the tube in a glass graduated cylinder, pressing the stop watch, and the REM buttons on my phone all at the same time.  Could have done with another hand is all.  Honestly now that mine is calibrated, I'd rather you work on better features.. lol

**Reply by rstrouse** [reply]
I assume you are looking for a function like below.   I still have some testing to do an will let you know when it is posted into the repo.
![image](https://user-images.githubusercontent.com/47839015/181593533-74d78cdc-a546-4512-bfff-6e753561eada.png)

**Comment by rstrouse** [comment]
I posted up the changes for the chem pump calibration.  You will find this on the hardware tab for the REM Chem controller.
![image](https://user-images.githubusercontent.com/47839015/181839426-555e14d8-de49-41f8-bdd1-254843252f4c.png)

When you press the button it will bring up the following dialog allowing you to run the pump for the specified number of seconds.
![image](https://user-images.githubusercontent.com/47839015/181839572-0d9eb3fe-8be0-492c-89c0-53ffc1517f98.png)

You will need to pull njsPC and dashPanel to get these changes.  You will also need to hard refresh your browser.
